spyder_kernels/customize/spydercustomize.py

```diff
@@ -753,6 +755,7 @@ def runcell(cellname, filename):
     ipython_shell.events.trigger('post_execute')
 
     ipython_shell.run_cell(cell_code)
+    namespace.pop('__file__')
```

Comment:
So __file__ only lives during the execution of the cell, not afterwards?


Just an FYI, this is contrary to the "run file" functionality, at least in 3.x

Reply:
Oh you are right, maybe I was running an older version.
I think it is ok to remove `__file__`, probably for the best anyway.

Reply:
Yes, I also think removing `__file__` after the command finishes is the best thing to do.